### PR TITLE
Show selected values in popup for SearchMultiSelectInput

### PR DESCRIPTION
**Pull request by frozenhelium**
* Fix Numeral exporting the class

### CODE REVIEW

File: components/View/Numeral/index.js [hunk 1]
```diff
@@ -90,7 +90,7 @@ const addNepaliSeparator = (num) => {
 /**
  * Numeral component for formatted numbers
  */
-export class NormalNumeral extends React.PureComponent {
```
Comment (AdityaKhatri) [inline components/View/Numeral/index.js]: @frozenhelium we need to export both NormalNumeral and wrapped numeral

File: components/Input/SearchMultiSelectInput/index.js [hunk 2]
```diff
@@ -346,11 +358,14 @@ class SearchMultiSelectInput extends React.PureComponent {
 
         const finalSearchValue = searchValue || '';
 
-        const filteredOptions = this.filterOptions(
+        const filteredOptions = this.getFilteredOptions(
             options,
             labelSelector,
+            keySelector,
             searchValue,
+            value,
```
Comment (AdityaKhatri) [inline components/Input/SearchMultiSelectInput/index.js]: Sending value to the filteredOptions causes a new list to be created whenever a value is selected which results in scroll jumps and unnecessary list calculations.

File: components/Input/SearchMultiSelectInput/index.js [hunk 3]
```diff
@@ -117,49 +121,62 @@ class SearchMultiSelectInput extends React.PureComponent {
...
+    ) => {
+        const optionsMap = listToMap(
+            options,
+            keySelector,
+            element => element,
+        );
+        const selectedOptions = value
+            .map(k => optionsMap[k])
+            .filter(isDefined)
+            .map(v => labelSelector(v));
```
Comment (tnagorra) [inline components/Input/SearchMultiSelectInput/index.js]:
```suggestion
            .map(labelSelector);
```

File: components/Input/SearchMultiSelectInput/index.js [hunk 4]
```diff
@@ -117,49 +121,62 @@ class SearchMultiSelectInput extends React.PureComponent {
...
+    findPlaceholderValue = memoize((
+        options,
+        labelSelector,
+        keySelector,
+        value = [],
+    ) => {
+        const optionsMap = listToMap(
+            options,
+            keySelector,
+            element => element,
```
Comment (tnagorra) [inline components/Input/SearchMultiSelectInput/index.js]:
```suggestion
            asIs,
```

File: components/Input/SearchMultiSelectInput/index.js [hunk 5]
```diff
@@ -346,12 +362,25 @@ class SearchMultiSelectInput extends React.PureComponent {
...
         const finalSearchValue = searchValue || '';
 
-        const filteredOptions = this.filterOptions(
-            options,
-            labelSelector,
-            searchValue,
-            maxDisplayOptions,
-        );
+        let filteredOptions = emptyFilteredOptions;
+        if (!searchValue || searchValue.length === 0) {
```
Comment (tnagorra) [inline components/Input/SearchMultiSelectInput/index.js]:
```suggestion
        if (isFalsyString(searchValue)) {
```

File: components/Input/SearchMultiSelectInput/index.js [hunk 6]
```diff
@@ -117,49 +121,62 @@ class SearchMultiSelectInput extends React.PureComponent {
...
     ) => {
-        if (!value || value.length === 0) {
-            return [];
-        }
-
         const newOptions = options
             .filter(
                 option => (
-                    value === undefined || caseInsensitiveSubmatch(labelSelector(option), value)
+                    searchValue === undefined
```
Comment (tnagorra) [inline components/Input/SearchMultiSelectInput/index.js]:
```suggestion
                    isFalsyString(searchValue)
```

**Review by tnagorra — APPROVED** [review]

**Pull request merged**